src/GestureAnimationDirector.vala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 elementary, Inc (https://elementary.io)
+ *           2020 José Expósito <jose.exposito89@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+public class Gala.GestureAnimationDirector : Object {
+    public bool running { get; set; default = false; }
+    public bool canceling { get; set; default = false; }
+
+    public signal void on_animation_begin (int percentage);
+    public signal void on_animation_update (int percentage);
+    public signal void on_animation_end (int percentage, bool cancel_action);
+
+    public delegate void OnBegin (int percentage);
+    public delegate void OnUpdate (int percentage);
+    public delegate void OnEnd (int percentage, bool cancel_action);
+
+    private Array<ulong> handlers;
```

Comment:
We tend to use Gee for lists / arrays and such. You could replace this type with e.g `Gee.ArrayList` and use all native features like `array[i]`, `array.clear ()` and `array.add` and `foreach (var id in array)`.